### ISSUE TITLE
Scikit metrics?

### ISSUE THREAD

**Issue by zbarni** [rcfduarte/nmsat#46]
While implementing the accuracy metric for the Legenstein project I realized that (some of) the sklearn.metrics don't behave as expected if they get matrices as arguments and not 1D vectors:

```python
binary_target = np.array([[1., 0., 1.], [1., 0., 0.]])
binary_output = np.array([[0., 0., 1.], [0., 0., 1.]])

# the library function
print "sklearn: ", met.accuracy_score(binary_target, binary_output)

# just to double check, manually
correct_predictions = 0
for row in range(binary_target.shape[0]):
    correct_predictions += met.accuracy_score(binary_target[row, :], binary_output[row, :], normalize=False)

print "should be: ", float(correct_predictions / binary_target.shape[0]
```

the output is

```
sklearn:  0.
should be:  0.5
```
There are 3 mismatches out of 6 values, so that's a 0.5. Seems like in this case `sklearn` is calculating the score row-by-row, and not for each element in the matrix. Not sure if this affects the label accuracy for other projects; depends if you give it matrices or only vectors..
